docs/plugins/restified-endpoints/how-to.mdx

```diff
@@ -0,0 +1,253 @@
+include raw strings here using `value` instead of `valueFromEnv` and passing the keys.
+
+:::
+
+Next, update the `subgraph.yaml` file to include the metadata file and the environment variables.
+
+```yaml
+kind: Subgraph
+version: v2
+definition:
+  name: globals
+  ...
+  includePaths:
+    ...
+    - restified-endpoints.hml
+  envMapping:
+    RESTIFIED_ENDPOINTS_URL:
+      fromEnv: RESTIFIED_ENDPOINTS_URL
+    M_AUTH_KEY:
+      fromEnv: M_AUTH_KEY
+```
+
+Finally, we need to add the environment variables to the `.env` file.
+
+```bash
+RESTIFIED_ENDPOINTS_URL="http://local.hasura.dev:8787"
+M_AUTH_KEY="your-strong-m-auth-key"
+```
+
+:::tip M-Auth Key
+
+The `hasura-m-auth` header is a custom header that is used to authenticate the requests to the allowlist plugin. You can
+use any strong key here to authenticate the plugin. DDN will automatically add this header to the requests to the
+plugin. Also, make sure to update the `src/config.ts` file (in step 5) with the same key.
+
+:::
+
+## Step 4. Create a new build for local development
+
+Create a new supergraph build.
+
+```bash
+ddn supergraph build local
+```
+
+Start the console for the local supergraph.
+
+```bash
+ddn console --local
+```
+
+You can now test the plugin by making a request to the RESTified GraphQL endpoints defined in the plugin's
+configuration.
+
+## Step 5. Update the plugin config
+
+Update the `src/config.ts` file with the RESTified GraphQL endpoints that you want to add.
+
+```typescript
+export const Config = {
+  graphqlServer: {
+    url: "http://localhost:3000/graphql",
+    headers: {
+      additional: {
+        "Content-Type": "application/json",
+      },
+      // Please make sure to forward the authentication headers here
+      forward: ["hasura_cloud_pat", "x-hasura-ddn-token"],
+    },
+  },
+  headers: {
+    "hasura-m-auth": "your-strong-m-auth-key",
+  },
+  restifiedEndpoints: [
+    ...,
+    {
+      path: "/v1/api/rest/users",
+      methods: ["GET", "POST"],
+      query: `
+        query MyQuery($limit: Int = 10, $offset: Int = 0) {
+          Users(limit: $limit, offset: $offset) {
+            Name
+          }
+        }
+      `,
+    }
+  ],
+};
+```
+
+:::info Hot reloading
+
+The local wrangler development server will automatically reload the plugin when you make changes to the code.
+
+:::
+
+## Step 6. Configure the plugin variables
+
+:::info Setup tracing
+
+To enable tracing for the plugin, you need to update the `wrangler.toml` file with the required configurations. If you
+don't want to enable tracing for the plugin, you can skip this step.
+
+:::
+
+In `restified-endpoints-plugin` directory, update the `wrangler.toml` file with the required configurations.
+
+```toml
+...
+[vars]
+OTEL_EXPORTER_OTLP_ENDPOINT = "https://gateway.otlp.hasura.io:443/v1/traces"
+OTEL_EXPORTER_PAT = "<PAT>"
+```
+
+Replace `<PAT>` with the Personal Access Token (PAT) for the Hasura Cloud account. You can generate this using the
```

Comment:
```suggestion
Replace `<PAT>` with the Personal Access Token (PAT) for the Hasura Cloud account. You can display this using the
```

Because it is pre-generated correct?

Reply:
Yes